packages/node_modules/@node-red/nodes/core/function/10-function.js

```diff
@@ -297,6 +302,7 @@ module.exports = function(RED) {
                 }
             });
             this.on("close", function() {
+                vm.runInContext(finText, context);
```

Comment:
This definitely does need error handling around it (in additional to my other comments on error handling) because we **must** clear and timeouts and intervals the node has created regardless of whether the finalize code has thrown an error or not.

I also suggest we skip the call entirely if there is no finalize function code provided.

Reply:
Fixed according to your suggestion.